allennlp/tests/data/dataset_readers/multiprocess_dataset_reader_test.py

```diff
@@ -72,6 +75,50 @@ def test_multiprocess_read(self):
         assert counts[("snakes", "are", "animals", ".", "N", "V", "N", "N")] == 100
         assert counts[("birds", "are", "animals", ".", "N", "V", "N", "N")] == 100
 
+    def test_multiprocess_read_partial_does_not_hang(self):
+        reader = MultiprocessDatasetReader(base_reader=self.base_reader, num_workers=4, output_queue_size=10)
+
+        all_instances = []
+
+        # Half of 100 files * 4 sentences / file
+        for _ in range(200):
+            instance = reader.read(self.identical_files_glob)
```

Comment:
this doesn't affect the test per se, but doesn't `reader.read` return an Iterable of instances? maybe just the variable name is misleading

Reply:
This was actually a bug. Nice catch!